kiota_authentication_azure/azure_identity_access_token_provider.py

```diff
@@ -45,11 +45,18 @@ async def get_authorization_token(self, uri: str) -> str:
         parsed_url = urlparse(uri)
         if not parsed_url.scheme == 'https':
             raise Exception("Only https is supported")
+        #async credentials
+        if inspect.iscoroutinefunction(self._credentials.get_token):
+            if self._options:
+                result = await self._credentials.get_token(*self._scopes, **self._options)
+            result = await self._credentials.get_token(*self._scopes)
+            await self._credentials.close()  #type: ignore
```

Comment:
[Their documentation](https://learn.microsoft.com/en-us/python/api/azure-core/azure.core.credentials_async.asynctokencredential?view=azure-python) is terrible but I'm wondering what close does and whether we need to call it after each authentication request or when we don't need the credential object anymore? (wondering whether we'll have the same issue as the http package where it stops working after the first request)

Reply:
Their PyPI documentation is a bit better
![image](https://user-images.githubusercontent.com/40166690/205071003-b07505ce-f4c5-40af-92ec-ca9cf7a3765c.png)

Reply:
that's what I suspected, they instruct to close the credentials when no longer needed, not after getting a token.
I believe the caller who owns the credentials object is responsible for that, not this library.
Thoughts?

Reply:
Ideally yes, but our setup currently does not allow us to reuse credentials since we are calling authentication_provider.authenticate_request(request_info) for each individual request. Therefore, closing the credential after getting a token does not affect subsequent requests.

Reply:
Different from the client scenario as we should be able to reuse a client for multiple requests.

Reply:
right, but are we able to get a token from a credentials after it's been closed?

Reply:
I think I now get your concern.
```py
async def get_creds():
    scopes = ['https://graph.microsoft.com/.default']
    credential = EnvironmentCredential()
    result = await credential.get_token(*scopes)
    await credential.close()
    result2 = await credential.get_token(*scopes)
    print(result2)
    
asyncio.run(get_creds())
```
The above prints the token in result 2 which implies you can actually can get a token after a credential has been closed. Also the reason we haven't faced the same issue as the client.

Reply:
to recap our discussion on the code, this could be a feature, this could be a bug. The documentation would point to closing only when we're done with the object, as opposed to right after using it. Please open an issue with them to clarify. But we can tackle that later since we're not aggravating the situation over existing.